examples/api/lib/material/menu_bar/menu_bar.0.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for MenuBar
+
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() => runApp(const MenuBarApp());
+
+enum MenuSelection {
+  about('About'),
+  showMessage('Show Message'),
+  hideMessage('Hide Message'),
+  colorMenu('Color Menu'),
+  colorRed('Red Background'),
+  colorGreen('Green Background'),
+  colorBlue('Blue Background'),
+  quit('Quit');
+
+  const MenuSelection(this.label);
+  final String label;
+}
+
+class MenuBarApp extends StatelessWidget {
+  const MenuBarApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'MenuBar Sample',
+      home: Scaffold(body: MyMenuBar()),
+    );
+  }
+}
+
+class MyMenuBar extends StatefulWidget {
+  const MyMenuBar({super.key});
+
+  @override
+  State<MyMenuBar> createState() => _MyMenuBarState();
+}
+
+class _MyMenuBarState extends State<MyMenuBar> {
+  bool get showingMessage => _showMessage;
+  bool _showMessage = false;
+  set showingMessage(bool value) {
+    if (_showMessage != value) {
+      setState(() {
+        _showMessage = value;
+      });
+    }
+  }
+
+  Color get backgroundColor => _backgroundColor;
+  Color _backgroundColor = Colors.red;
+  set backgroundColor(Color value) {
+    if (_backgroundColor != value) {
+      setState(() {
+        _backgroundColor = value;
+      });
+    }
+  }
+
+  void _activate(MenuSelection selection) {
+    switch (selection) {
+      case MenuSelection.about:
+        showAboutDialog(
+          context: context,
+          applicationName: 'MenuBar Sample',
+          applicationVersion: '1.0.0',
+        );
+        break;
+      case MenuSelection.showMessage:
+        showingMessage = true;
+        break;
+      case MenuSelection.hideMessage:
+        showingMessage = false;
+        break;
+      case MenuSelection.quit:
```

Comment:
We may want to think carefully about how desktop applications should "quit". Most toolkits provide some support for orderly shutdown, including bugging the user about unsaved changes etc.  What if we left this aspect of the demo out - users could still just close the window.